assets/src/dashboard/app/views/storyAnimTool/timeline/index.js

```diff
@@ -0,0 +1,255 @@
+        0
+      ),
+    [animations]
+  );
+
+  const handleAnimationSubmit = useCallback(
+    (e) => {
+      e.preventDefault();
+
+      if (!isAnimationSaveable) {
+        return;
+      }
+
+      const animation = {
+        id: e.target.id.value || uuidv4(),
+        type: e.target.type.value,
+        duration: parseInt(e.target.duration.value),
+        delay: parseInt(e.target.delay.value),
+      };
+
+      e.target.reset();
+
+      onAddOrUpdateAnimation(animation);
+    },
+    [isAnimationSaveable, onAddOrUpdateAnimation]
+  );
+
+  const handleToggleTargetSelect = useCallback(
+    (e) => {
+      e.preventDefault();
+      onToggleTargetSelect(!isElementSelectable);
+    },
+    [isElementSelectable, onToggleTargetSelect]
+  );
+
+  const handleCancelClick = useCallback(
+    (e) => {
+      e.preventDefault();
+      onAnimationSelect('');
+    },
+    [onAnimationSelect]
+  );
+
+  const handleDeleteClick = useCallback(
+    (animationId) => {
+      if (window.confirm('Are you sure you want to delete this animation?')) {
+        onAnimationDelete(animationId);
+      }
+    },
+    [onAnimationDelete]
+  );
+
+  const handleInputFocus = useCallback((e) => e.target.select(), []);
+
+  useEffect(() => {
+    onToggleTargetSelect(false);
+  }, [activeAnimation, onToggleTargetSelect]);
+
+  return (
+    <>
+      <p>{`Total animation duration for this page: ${totalDuration}ms`}</p>
+      <Container>
+        <AnimationList>
+          {animations.map((animation) => (
+            <TimelineAnimation key={animation.id}>
+              <TimelineLabel>
+                <DeleteButton
+                  title="Delete Animation"
+                  onClick={(e) => {
+                    e.preventDefault();
+                    handleDeleteClick(animation.id);
+                  }}
+                >
+                  <DeleteIcon />
+                </DeleteButton>
+                <LabelButton
+                  isActive={activeAnimation.id === animation.id}
+                  onClick={() => onAnimationSelect(animation.id)}
+                >
+                  {animation.type}
+                </LabelButton>
+              </TimelineLabel>
+              <TimelineBarContainer>
+                <TimelineBar
+                  title={`${animation.duration}ms`}
+                  onClick={() => onAnimationSelect(animation.id)}
+                  {...getOffsetAndWidth(
+                    totalDuration,
+                    animation.duration,
+                    animation.delay
+                  )}
+                />
+              </TimelineBarContainer>
+            </TimelineAnimation>
+          ))}
+        </AnimationList>
+        <AnimationPanel>
+          <form onSubmit={handleAnimationSubmit}>
+            <FormField>
+              <label>{'Animation Type'}</label>
+              <select
+                key={`animation-type: ${activeAnimation.id}`}
```

Comment:
Setting a key that is unique per animation lets us change the `defaultValue` of the control as we change active animations without having to make the selects and inputs controlled by react.  It's a hack so I didn't have to keep each field in state and just let the form do its thang while still being able to pre-fill the form as needed.  For a prod level tool I wouldn't do this, but for this quick and dirty dev-tool I'm okay with it :)